bluejay-validator/src/executable/operation/visitor.rs

```diff
@@ -47,23 +58,43 @@ pub trait Visitor<'a, E: ExecutableDocument, S: SchemaDefinition, V: VariableVal
     fn visit_variable_definition(&mut self, variable_definition: &'a E::VariableDefinition) {}
 }
 
+pub trait VisitorExtraInfo<'a, E: ExecutableDocument, S: SchemaDefinition, V: VariableValues> {
+    type ExtraInfo;
+}
+
+impl<
+        'a,
+        E: ExecutableDocument,
+        S: SchemaDefinition,
+        V: VariableValues,
+        T: Visitor<'a, E, S, V>,
+    > VisitorExtraInfo<'a, E, S, V> for T
+{
+    type ExtraInfo = T::ExtraInfo;
+}
+
 macro_rules! impl_visitor {
     ($n:literal) => {
         seq_macro::seq!(N in 0..$n {
             #[warn(clippy::missing_trait_methods)]
             impl<'a, E: ExecutableDocument, S: SchemaDefinition, V: VariableValues, #(T~N: Visitor<'a, E, S, V>,)*> Visitor<'a, E, S, V> for (#(T~N,)*) {
+                type ExtraInfo = (#(<T~N as VisitorExtraInfo<'a, E, S, V>>::ExtraInfo,)*);
```

Comment:
Couldn't this just be?
```suggestion
                type ExtraInfo = (#(<T~N as Visitor<'a, E, S, V>>::ExtraInfo,)*);
```